docs/develop/python/message-passing.mdx

```diff
@@ -27,155 +27,179 @@ tags:
   - dynamic-handlers
 ---
 
-This page shows how to do the following:
+Here is an example Workflow Definition featuring Query, Signal, and Update handlers, and an Update validator.
+You can view this as a runnable sample at [hello/hello_message_passing.py](https://github.com/temporalio/samples-python/blob/dan/dan/docs_message_passing.py).
```

Comment:
Right, I did consider that. However, the directory would need a rename since my sample very much covers Query also. Do you think renaming your directory, e.g. to `message_passing`, might make sense? I don't have a strong opinion there -- I can also see that having a place to discuss async handlers and the pitfalls thereof has value.

No particular love for `hello/`, but I'm going to push for maintaining single-file structure for simple toy samples like this, so that they can simply be run by `python the_example.py` from within a virtualenv.

Reply:
Same, no opinion. We can be fast/loose w/ sample structure because we can change as needed.

Reply:
> I'm going to push for maintaining single-file structure for simple toy samples like this, so that they can simply be run by `python the_example.py` from within a virtualenv.

Actually, no, I have changed my mind. It's very hard for users to understand what's happening when there are errors if they are running a single-file sample with starter and worker stdin & stderr all going to a single terminal. I'm splitting it into `starter/worker/workflow` and am going to rename the directory from `updates_and_signals => message_passing`:

```
message_passing
├── introduction
└── safe_message_handlers
```